lib/currency_conversion/update_worker.ex

```diff
@@ -27,22 +27,57 @@ defmodule CurrencyConversion.UpdateWorker do
 
     refresh_interval = Keyword.get(opts, :refresh_interval, @default_refresh_interval)
 
-    case refresh(opts) do
-      :ok -> {:ok, opts, refresh_interval}
-      {:error, binary} -> {:stop, {:error, binary}}
+    case refresh_interval do
+      :manual ->
+        {:ok, opts, refresh_interval}
+
+      interval ->
+        case refresh(opts) do
+          :ok -> {:ok, opts, refresh_interval}
+          {:error, binary} -> {:stop, {:error, binary}}
+        end
     end
   end
 
   @impl GenServer
   def handle_info(:timeout, opts) do
     refresh_interval = Keyword.get(opts, :refresh_interval, @default_refresh_interval)
 
-    case refresh(opts) do
-      :ok -> {:noreply, opts, refresh_interval}
-      {:error, binary} -> {:stop, {:error, binary}}
+    case refresh_interval do
+      :manual ->
+        {:reply, :ok, opts}
+
+      interval ->
+        case {refresh(opts), refresh_interval} do
+          {:ok, :manual} -> {:ok, opts}
+          {:ok, _} -> {:ok, opts, refresh_interval}
+          {{:error, binary}, _} -> {:stop, {:error, binary}}
+        end
     end
   end
 
+  @impl GenServer
+  def handle_call(:refresh, _from, opts) do
+    refresh_interval = Keyword.get(opts, :refresh_interval, @default_refresh_interval)
+
+    case refresh_interval do
```

Comment:
Also in case of `manual`, the `refresh function should be called. The whole case should look like this therefore:

```elixir
case {refresh(opts), refresh_interval} do
    {:ok, :manual} -> {:reply, :ok, opts}
    {:ok, _} -> {:reply, :ok, opts, refresh_interval}
    {{:error, binary}, :manual} -> {:reply, {:error, binary}, opts}
    {{:error, binary}, _} -> {:reply, {:error, binary}, opts, refresh_interval}
end
```

Reply:
Just want you to confirm perhaps, but `refresh(opts)` would always do `load` right now no? There is no `:manual` handling in the refresh itself since that one has to do just refresh when called.